### DISCUSSION TITLE
withProps should take partial props instead of all props

### DISCUSSION THREAD

**Discussion by arvind0598** [mantinedev/mantine#7207] [Feature requests]
I've been playing around with creating generic (opt-in) components that the rest of the team can use. Using the full props instead of partial is an issue for situations like this:

```tsx
export const AppDrawer = Drawer.withProps({
  position: "right",
  offset: 8,
  radius: "md",
  styles: {
    content: {
      display: "flex",
      flexDirection: "column",
    },
    body: {
      flexGrow: 1,
      display: "flex",
      flexDirection: "column",
    },
  },
});
```

This causes an error saying `... is missing the following properties from type 'DrawerProps': opened, onClose`. I don't think I should be providing those props here at all. For now, I'm able to work around it by just forcing all drawers to follow this via the createTheme, but I wanted to understand the rationale here so that I can use it for stuff like variants later.

**Comment by rtivital** [comment]
Okay, I'll check that
